packages/avro-ts/src/types/primitive.ts

```diff
@@ -24,7 +24,17 @@ export const isPrimitiveType = (type: Schema): type is schema.PrimitiveType =>
   type === 'float' ||
   type === 'double' ||
   type === 'bytes' ||
-  type === 'string';
+  type === 'string' ||
+  (typeof type === 'object' &&
+    'type' in type &&
+    typeof type.type === 'string' &&
+    isPrimitiveType(type.type));
```

Comment:
Instead of putting this in the primitive type, we could instead use a totally new type. I've noticed that there is something called "NamedType" in the Avro typescript types, that denotes that.

Maybe we can use it to our advantage? 

We can create something like
```typescript
import { schema, Schema } from 'avsc';
import { Convert } from '../types';
import { convertPrimitiveType, isPrimitiveType } from './primitive';

export const isNamedType = (type: Schema): type is schema.NamedType =>
  typeof type === 'object' && 'type' in type && isPrimitiveType(type.type);

export const convertNamedType: Convert<schema.NamedType> = (context, schema) =>
  convertPrimitiveType(context, schema.type);
```

Which would naturally be used in `convert.ts` right before the `convertPrimitiveType`.
```typescript
  } else if (isFixedType(type)) {
    return convertFixedType(context, type);
  } else if (isNamedType(type)) {
    return convertNamedType(context, type);
  } else if (isPrimitiveType(type)) {
    return convertPrimitiveType(context, type);
```
This way we should be able to handle all named types, not just the string ones you have.

What do you think?

Reply:
Two things:
- I think that it should already be able to handle more than just string primitives. But I confess, I haven't tested it with enums or others. Maybe I should 😄 
- I thought about extracting everything in a separate type-handler. I decided that I want to be minimal intrusive in your codebase 🙂  But extracting it is also possible.

How do we proceed? Should I adjust the merge request or do you want to make the changes yourself?

Reply:
🤦🏻  I'm an idiot sometimes. I keep trying to test with enums but enums aren't primitives... Damn you brain...
Anyway. I made the changes. Actually, just copied them from your request 😄  and added a second avsc sample - just to be sure.